apis/contexts.go

```diff
@@ -85,14 +85,21 @@ func IsInUpdate(ctx context.Context) bool {
 	return ctx.Value(inUpdateKey{}) != nil
 }
 
-// IsInStatusUpdate checks whether the context is an Update.
-func IsInStatusUpdate(ctx context.Context) bool {
+// IsInSubResourceUpdate returns the subresource being updated or "" if there
+// is no subresource that's being updated. Examples are "status" for Status
+// updates, or "scale" for scaling Deployment.
+func IsInSubResourceUpdate(ctx context.Context) string {
```

Comment:
Calling is `Is...` feels weird returning a string.  Maybe `GetUpdatedSubresource`?